### ISSUE TITLE
Using `anomalies` preprocessor with the reference period outside of time range of interest

### ISSUE THREAD

**Issue by malininae** [ESMValGroup/ESMValCore#2308]
Partially related to #1915, but not related to dask usage. Partially related to #2304. 

By using [this recipe](https://github.com/malininae/ESMValTool/blob/extremes_bc/esmvaltool/recipes/malinina23/recipe_bc_extremes_txx.yml) there are issues with anomalies preprocessor. For example, in that case I'm interested in three different periods 1850-1900, 2011-2030 and 2081-2100, however, I'm interested not in the absolute values, but rather in the anomalies with the reference period from 1940 to 1969. Since currently the anomalies preprocessor extracts the reference period from the provided cube ([see here](https://github.com/ESMValGroup/ESMValCore/blob/b72da76b04c6dbf9bdb94a18bd91192e9c390321/esmvalcore/preprocessor/_time.py#L866)) one needs to load the full 1850-1969, 1940-2020 and 1940-2100 cubes; and after the anomalies are calculated, use `extract_timerange` for 1850-1900, 2011-2030 and 2081-2100 respectively. While it worked OK for that particular recipe, it is getting unnecessarily computationally expensive, if using for example daily HighResMIP. Furthermore, one can't use `start_year`/`end_year`/`timerange` keywords from `cfg` since they apply for the full timeseries. 

I can even help fixing it, but don't know what would be the best tactic to solve that issue, change `anomalies` that way that it is OK to load two partial cubes? Not to use the preprocessor `anomaly` and do like I've done [here](https://github.com/malininae/ESMValTool/blob/extremes_bc/esmvaltool/recipes/malinina23/recipe_bc_extremes_tx3x.yml) by creating a separate group and subtracting anomalies in the diagnostic? If the latter than I would suggest raising a warning in the `anomalies` preprocessor. Tagging @bouweandela since we touched on that issue in the above mentioned issues.

**Comment by bouweandela** [comment]
If I understand the issue here correctly, the most efficient way to compute this would probably be to only compute the reference and the periods of interest using variable groups in the recipe and then do the final anomaly computation in the diagnostic. To facilitate that, we could consider making it possible to pass a cube as the `reference` argument to the `anomalies` preprocessor function. That should also solve #2016. @schlunma Do you think this could work?

**Comment by schlunma** [comment]
Yeah, I think that's a good idea! I did something similar to the bias preprocessor in https://github.com/ESMValGroup/ESMValCore/pull/2183.

**Comment by malininae** [comment]
Thanks all for the comments, I will look at what @schlunma did and will try to implement passing a cube as a reference (not earlier than within a month in the best case scenario). I'm also proposing to add mentioning to the `anomalies` documentation that the anomalies are calculated from the time period within the cube. It's not obvious for the outsider in particular if one expects it to be used even more in CMIP7.

**Comment by bouweandela** [comment]
A pull request to add that would be very welcome.
